cypress-io/cypress
# fix: Passing cohort through to SelectCloudProjectModal

- Closes https://github.com/cypress-io/cypress/issues/24068

### User facing changelog


n/a

### Additional details


The `cloudProjectCreate` mutation was updated to support a new cohort field before the cohort functionality was implemented.  The `SelectCloudProjectModal` that calls this mutation was stubbed out to pass an empty string.  When the cohort functionality was implemented, the cohort being selected in `SpecsListBanners` was not passed through the `CreateProjectBanner` to the `SelectCloudProjectModal`.

### Steps to test


NOTE: The `state.json` file for a project can be found by going to "View App Data" in the Developer Tools menu of the Cypress launchpad and then going to the appropriate project directory under "projects"

#### Setup the app to get it to show the "CreateProjectBanner"
1. Use a project that has been opened for longer than 4 days.
   * The timestamp for "firstOpened" in the state.json file for the project can be adjusted if needed.  
2. Use a project that does not have a project connected
   * There should not be a "projectId" in the `cypress.config.{ts,js}` file 
3. Make sure the project has not dismissed the banners or else they will not appear
   *  To start the banner settings from the initial state, delete the "banners" section in the `state.json` file for the project

#### Testing steps
1. Open the Cypress app in open mode and launch the browser into either e2e or component testing mode
2. Log in to the cloud with a user that has at least one organization or create a new organization when prompted
3. On the Specs list page, you should be presented with the "CreateProjectBanner"
4. Select the call to action button that says "Connect project"
5. Use the developer tools to watch for the graphql mutation [See screenshot below]
  * Open the developer tools for the browser
  * Go to the network tab
  * Filter the connections based on websockets ("WS" in the filters)
  * Select the websocket connection that says `?EIO=4&transport=websocket` - refresh the browser if you do not see it
  * Go to the "Messages" tab for the connection and filter on "CreateCloudProject"
6. Back in the modal window in the app, choose to create a new project, not to connect to an existing project
7. After you click the button to create the project, go back to the Developer Tools window to verify the mutation
  * The filtered messages tab should now have a message in it
  * Scroll to the right in the result window to look for the arguments to the mutation.
  * Verify that the cohort has a value
  * Verify that this value matches the cohort value stored in the `cache` file under the key `COHORTS. aci_082022_connectProject` - The cache file is located in the directory accessed via "View App Data" in the launchpad "Developer Tools" window
![Devtools_filter_websocket](https://user-images.githubusercontent.com/1702361/193692327-d6bba6c1-d796-4761-8f38-e87cb3cd05f5.png)

### How has the user experience changed?


N/A

### PR Tasks


- [y ] Have tests been added/updated?
- [n/a] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [n/a] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [n/a] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
